### PR TITLE
Backend user tests fixes

### PR DESCRIPTION
**Pull request by byroncjy**
Adds profile picture route tests
Updates user route tests (user and account info) according to our backend & schema changes

Note that the new image file in test/public/ is for the upload picture test

Consolidated into validUserId and invalidUserId

### CODE REVIEW

**Review by SivanC — COMMENTED** [review]
Looks good, still some small changes to make. I think that the failure ids should be consolidated and also should be in a valid id format, otherwise I think it will flag that first instead of completing the request

File: back-end/test/user.test.js [hunk 1]
```diff
@@ -8,12 +9,11 @@ const nock = require('nock')
 // routes
 const router = require('../app')
 
-
 // tests related to sending requests about a users
 describe('User Routes', function () {
-  // tests that result in success will ask for id 0, and tests that result in
+  // tests that result in success will ask for valid id, and tests that result in
   // failure will ask for id 1
```
[the remark this answers is not in the document]
Comment (byroncjy) [reply back-end/test/user.test.js:15]: fixed

File: back-end/test/user.test.js [hunk 2]
```diff
@@ -8,12 +9,11 @@ const nock = require('nock')
...
-  const successId = 0
+  const successId = '6562c186a4a586c6e19a4eef'
```
[the remark this answers is not in the document]
Comment (byroncjy) [reply back-end/test/user.test.js:16]: added comment

File: back-end/test/user.test.js [hunk 3]
```diff
@@ -66,11 +66,11 @@ describe('User Routes', function () {
             email: 'email',
             first_name: 'first_name',
             last_name: 'last_name',
-            url_picture: 'url_picture'
+            profile_picture: 'profile_picture'
           })
           .end((err, res) => {
             expect(res).to.have.status(200)
-            expect(res.body).to.be.a('object').that.has.all.keys('message', 'updatedUserData')
+            expect(res.body).to.be.a('object').that.has.all.keys('message')
```
[the remark this answers is not in the document]
Comment (byroncjy) [reply back-end/test/user.test.js:73]: removed

File: back-end/test/user.test.js [hunk 4]
```diff
@@ -140,15 +142,80 @@ describe('User Routes', function () {
...
     })
   })
+  // Profile picture routes: /profilePicture, /uploadPicture
+  describe('Profile Picture', function () {
+    describe('GET profilePicture', function () {
+      describe('Successful GET', function () {
+        it('should respond with an HTTP 200 status and profile picture data', function (done) {
+          chai
+            .request(router)
+            .get(`/users/${successId}/profilePicture`)
```
[the remark this answers is not in the document]
Comment (byroncjy) [reply back-end/test/user.test.js:165]: yes, it's the route for getting only the profile picture

File: back-end/test/user.test.js [hunk 5]
```diff
@@ -81,10 +81,10 @@ describe('User Routes', function () {
           .request(router)
           .put(`/users/${successId}`)
```
[the remark this answers is not in the document]
Comment (byroncjy) [reply back-end/test/user.test.js]: renamed/consolidated into 2 ids

File: back-end/test/user.test.js [hunk 6]
```diff
@@ -140,15 +142,80 @@ describe('User Routes', function () {
...
-            { platform: 'bad_platform_without_info'}
-          ])
+          .send({
+            platforms: [
+              { wrong_name: 'bad_platform', wrong_info: 'johndoe254' }
+            ]
+          })
           .end((err, res) => {
-            expect(res).to.have.status(400)
+            expect(res).to.have.status(500)
```
[the remark this answers is not in the document]
Comment (byroncjy) [reply back-end/test/user.test.js:138]: fixed

**Comment by byroncjy** [comment]
Also added another test (no file attached) for uploadPicture PUT

**Pull request merged**